azuki-script-generation/src/main/kotlin/com/anaplan/engineering/azuki/script/generation/ScriptGenerationTestHelper.kt

```diff
@@ -0,0 +1,44 @@
+package com.anaplan.engineering.azuki.script.generation
+
+import com.anaplan.engineering.azuki.core.parser.ScenarioParser
+import com.anaplan.engineering.azuki.core.parser.ScenarioParsingContext
+import com.anaplan.engineering.azuki.core.parser.SimpleScenarioParser
+import com.anaplan.engineering.azuki.core.scenario.BuildableScenario
+import com.anaplan.engineering.azuki.core.system.ActionFactory
+import org.junit.Assert
+import org.slf4j.LoggerFactory
+
+open class ScriptGenerationTestHelper<S: BuildableScenario<AF>, AF: ActionFactory>(
+    private val generator: ScriptGenerator<AF, *, *, *, *>,
+    private val parser: ScenarioParser<S> = SimpleScenarioParser(),
+) {
+    /**
+     * Verify script correctness by generating, parsing and regenerating.
+     *
+     * If there are no exceptions then all constructs in initial scenario will have been supported. If the two scripts
+     * generated are identical then we know the initial scenario and parsed scenario are semantically equivalent.
+     *
+     * Note that, there may be some differences between the initial dsl and that generated, but they will produce the
+     * same actions, checks, and queries.
+     *
+     * printScript will print the initially generated script to assist debugging
```

Comment:
I don't think this comment is valid anymore, but I didn't want to edit it in case I was just reading it wrong?